sdk/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -134,11 +136,46 @@ public String getEndpoint() {
      * endpoint.
      *
      * @param endpoint the custom endpoint to send report to
+     * @deprecated use {@link com.bugsnag.android.Configuration#setEndpoints(String, String)}
      */
+    @Deprecated
     public void setEndpoint(String endpoint) {
         this.endpoint = endpoint;
     }
 
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://session.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoint.
+     *
+     * Please note that it is recommended that you set both endpoints. If the notify endpoint is
+     * missing, an exception will be thrown. If the session endpoint is missing, a warning will be
+     * logged and sessions will not be sent automatically.
+     *
+     * @param notify the notify endpoint
+     * @param sessions the sessions endpoint
+     *
+     * @throws IllegalArgumentException if the notify endpoint is empty or null
+     */
+    @SuppressWarnings("ConstantConditions")
+    public void setEndpoints(@NonNull String notify, @NonNull String sessions)
+        throws IllegalArgumentException {
+        boolean invalidNotify = TextUtils.isEmpty(notify);
+        invalidSessionsEndpoint = TextUtils.isEmpty(sessions);
+
+        if (invalidNotify) {
+            throw new IllegalArgumentException("Notify endpoint cannot be empty or null.");
+        }
+        if (invalidSessionsEndpoint) {
+            Logger.warn("The session tracking endpoint has not been set. "
+                + "Session tracking is disabled");
+            autoCaptureSessions = false;
+        }
+        this.endpoint = notify;
+        this.sessionEndpoint = sessions;
```

Comment:
if `invalidSessionsEndpoint` is true, then shouldn't this be set to null? Whether to send a session could then be determined by whether the endpoint is null rather than adding a new property.

It would also reduce the odds of unexpected behavior if `Bugsnag.setAutoCaptureSessions()` is called after configuration.